tests/Tests/ORM/Mapping/ClassMetadataTest.php

```diff
@@ -109,9 +109,11 @@ public function testClassMetadataInstanceSerialization(): void
         self::assertInstanceOf(ReflectionClass::class, $cm->reflClass);
         self::assertEquals(CmsUser::class, $cm->name);
         self::assertEquals('UserParent', $cm->rootEntityName);
-        self::assertEquals([One::class, Two::class, Three::class], $cm->subClasses);
+        self::assertNotEquals([One::class, Two::class, Three::class], $cm->subClasses);
+        self::assertEquals(['One', 'Two', 'Three'], $cm->subClasses);
```

Comment:
It looks like you read https://github.com/doctrine/orm/issues/11208 and targeted 4.0.x since this is a breaking change. Good.
Breaking changes must be documented in `UPGRADE.md` though, and an upgrade path should allow users to opt in to the new behavior on 3.x (and the old behavior should be deprecated in favor of the new one).